common/network/IPV6Address.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * IPV6Address.cpp
+ * An IPV6 address
+ * Copyright (C) 2023 Peter Newman
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif  // HAVE_CONFIG_H
+
+#ifdef HAVE_SYS_SOCKET_H
+#include <sys/socket.h>  // Required by FreeBSD
+#endif  // HAVE_SYS_SOCKET_H
+#ifdef HAVE_ARPA_INET_H
+#include <arpa/inet.h>
+#endif  // HAVE_ARPA_INET_H
+#ifdef HAVE_NETINET_IN_H
+#include <netinet/in.h>  // Required by FreeBSD
+#endif  // HAVE_NETINET_IN_H
+
+#include <assert.h>
+#include <math.h>
+#include <stdint.h>
+#include <limits>
+#include <string>
+
+#include "common/network/NetworkUtilsInternal.h"
+#include "ola/Logging.h"
+#include "ola/network/IPV6Address.h"
+#include "ola/network/NetworkUtils.h"
+
+namespace ola {
+namespace network {
+
+using std::string;
+
+IPV6Address::IPV6Address(const uint8_t *address) {
+  // TODO(Peter): Deal with any network byte order conversion?
+  memcpy(&m_address.s6_addr[0], address, sizeof (struct in6_addr));
+}
+
+bool IPV6Address::operator<(const IPV6Address &other) const {
+  // TODO(Peter): Deal with any network byte order conversion?
+  // Stored in network byte order, so convert to sort appropriately
+  return (memcmp(&m_address.s6_addr[0],
+          &other.m_address.s6_addr[0],
+          sizeof (struct in6_addr)) < 0);
+}
+
+bool IPV6Address::operator>(const IPV6Address &other) const {
+  // TODO(Peter): Deal with any network byte order conversion?
```

Comment:
can this TODO comment be removed? The next line seems to clarify it?

Reply:
As above

Reply:
As above again.